test/constraint/spec.test.ts

```diff
@@ -765,6 +765,63 @@ describe('constraints/spec', () => {
     });
   });
 
+  describe('omitNonLinearScaleTypeWithStack', () => {
+    it('should return false for stack with non linear scale type', () => {
+      [ScaleType.LOG, ScaleType.ORDINAL, ScaleType.POW, ScaleType.QUANTILE, ScaleType.QUANTIZE,
+       ScaleType.SQRT, ScaleType.TIME, ScaleType.UTC].forEach((scaleType) => {
+        const specM = buildSpecQueryModel({
+          mark: Mark.BAR,
+          encodings: [
+            {channel: Channel.X, field: 'A', type: Type.QUANTITATIVE, scale: {type: scaleType}, aggregate: AggregateOp.SUM},
+            {channel: Channel.Y, field: 'B', type: Type.NOMINAL},
+            {channel: Channel.COLOR, field: 'C', type: Type.NOMINAL}
+          ]
+        });
+        assert.isFalse(SPEC_CONSTRAINT_INDEX['omitNonLinearScaleTypeWithStack'].satisfy(specM, schema, DEFAULT_QUERY_CONFIG));
+      });
+    });
+
+    it('should return true for stack with linear scale type', () => {
+      const specM = buildSpecQueryModel({
+        mark: Mark.BAR,
+        encodings: [
+          {channel: Channel.X, field: 'A', type: Type.QUANTITATIVE, scale: {type: ScaleType.LINEAR}, aggregate: AggregateOp.SUM},
+          {channel: Channel.Y, field: 'B', type: Type.NOMINAL},
+          {channel: Channel.COLOR, field: 'C', type: Type.NOMINAL}
+        ]
+      });
+      assert.isTrue(SPEC_CONSTRAINT_INDEX['omitNonLinearScaleTypeWithStack'].satisfy(specM, schema, DEFAULT_QUERY_CONFIG));
+    });
+
+    it('should return true if color uses a non-linear scale when it is mapped to a non-X or non-Y channel that is aggregate', () => {
+      const specM = buildSpecQueryModel({
+        mark: Mark.BAR,
+        encodings: [
+          {channel: Channel.X, field: 'A', type: Type.QUANTITATIVE, aggregate: AggregateOp.SUM},
+          {channel: Channel.Y, field: 'B', type: Type.NOMINAL},
+          {channel: Channel.COLUMN, field: 'C', type: Type.NOMINAL, aggregate: AggregateOp.ARGMAX},
+          {channel: Channel.COLOR, field: 'C', type: Type.NOMINAL, scale: {type: ScaleType.ORDINAL}},
```

Comment:
I don't see why this test case is making any difference from above since nominal always use ORDINAL scale already.  

What you actually should test is mapping quantitative field to color with non-linear scale (e.g., pow)